doc: add Networking and security observability with Hubble guide

michi-covalent opened the pull request:
Signed-off-by: Michi Mutsuzaki <michi@isovalent.com>

rolinh committed:
doc: split kind guide sections into files to be reused by other guides

This will be used for a new Getting Started with Hubble guide.
While here, fix a few typos.

Co-authored-by:: Michi Mutsuzaki <michi@isovalent.com>
Signed-off-by: Michi Mutsuzaki <michi@isovalent.com>
Signed-off-by: Robin Hahling <robin.hahling@gw-computing.net>

rolinh committed:
doc: add Networking and security observability with Hubble guide

Co-authored-by: Michi Mutsuzaki <michi@isovalent.com>
Signed-off-by: Michi Mutsuzaki <michi@isovalent.com>
Signed-off-by: Robin Hahling <robin.hahling@gw-computing.net>

errordeveloper commented on Documentation/gettingstarted/hubble.rst:
```diff
@@ -0,0 +1,143 @@
+
+.. parsed-literal::
+
+  docker pull cilium/cilium:|IMAGE_TAG|
```
If this would be omitted, wouldn't `kind load` just pull it?

gandro replied:
Nope: 
```console
$ kind load docker-image cilium/cilium:v1.8                                                                                                                                                                                                                                              
ERROR: image: "cilium/cilium:v1.8" not present locally
```

errordeveloper commented on Documentation/gettingstarted/hubble.rst:
```diff
@@ -0,0 +1,143 @@
+
+.. parsed-literal::
+
+  docker pull cilium/cilium:|IMAGE_TAG|
```
It would make sense to have all of the images here - `cilium/cilium`, `cilium/cilium-operator-generic` `cilium/hubble-relay` and `cilium/startup-script`, but that is a long list - how much difference does this really make if no image were pre-loaded?

gandro replied:
These are not pulled by every node (except maybe startup-script). And they are also much smaller.

rolinh replied:
It would actually be a waste of resources to pre-load images that are not deployed on every node.

gandro committed:
doc: Add section about Hubble UI

This adds a quick how-to for Hubble UI to the Hubble Getting Started
Guide. It deploys the Star Wars demo app from the policy tutorials.

Signed-off-by: Sebastian Wicki <sebastian@isovalent.com>

rolinh committed:
doc: add section about Hubble Relay to the Hubble getting started guide

Signed-off-by: Robin Hahling <robin.hahling@gw-computing.net>
